From 342caf3345f581c88054de3d78c3f173d7a08aad Mon Sep 17 00:00:00 2001
From: pietfried <pietgoempel@gmail.com>
Date: Mon, 5 Aug 2024 13:45:55 +0200
Subject: [PATCH] Registered time_sync_callback in OCPP201 module

Signed-off-by: pietfried <pietgoempel@gmail.com>
---
 modules/OCPP201/OCPP201.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/modules/OCPP201/OCPP201.cpp b/modules/OCPP201/OCPP201.cpp
index fe92ae85..6fe989a2 100644
--- a/modules/OCPP201/OCPP201.cpp
+++ b/modules/OCPP201/OCPP201.cpp
@@ -529,6 +529,10 @@ void OCPP201::ready() {
         return;
     };
 
+    callbacks.time_sync_callback = [this](const ocpp::DateTime& current_time) {
+        this->r_system->call_set_system_time(current_time.to_rfc3339());
+    };
+
     const auto sql_init_path = this->ocpp_share_path / SQL_CORE_MIGRATIONS;
 
     std::map<int32_t, int32_t> evse_connector_structure = this->get_connector_structure();
-- 
2.34.1

