From 02347bbda0e785724a39ad5ae9ec55c6f97fa603 Mon Sep 17 00:00:00 2001
From: Fabian Hartung <fabian.hartung@chargebyte.com>
Date: Thu, 13 Mar 2025 15:43:49 +0100
Subject: [PATCH 1/2] Evse15118D20: Add pre-charge started signal handling

Signed-off-by: Fabian Hartung <fabian.hartung@chargebyte.com>
---
 modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp b/modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp
index 876845b7..b2bf0a70 100644
--- a/modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp
+++ b/modules/Evse15118D20/charger/ISO15118_chargerImpl.cpp
@@ -267,6 +267,9 @@ iso15118::session::feedback::Callbacks ISO15118_chargerImpl::create_callbacks()
     callbacks.signal = [this](feedback::Signal signal) {
         using Signal = feedback::Signal;
         switch (signal) {
+        case Signal::PRE_CHARGE_STARTED:
+            publish_start_pre_charge(nullptr);
+            break;
         case Signal::CHARGE_LOOP_STARTED:
             publish_current_demand_started(nullptr);
             break;
-- 
2.34.1

